syntax = "proto3";

package debate_event;

import "google/protobuf/timestamp.proto";

// Individual event payloads
message CreateDebate {
    string user_id = 1;
    string debate_topic = 2;
}

message ClearDebates {
    string user_id = 1;
}

message DeleteDebate {
    string user_id = 1;
    string debate_id = 2;
}

message EnterDebate {
    string user_id = 1;
    string debate_id = 2;
}

message GoHome {
    string user_id       = 1;
}

// enum
enum EventKind {
    EVENT_KIND_UNSPECIFIED = 0;
    CREATE_DEBATE          = 1;
    CLEAR_DEBATES          = 2;
    DELETE_DEBATE          = 3;
    ENTER_DEBATE           = 4;
    GO_HOME                = 5;
}

// main message
message DebateEvent {

  string id = 1;
  google.protobuf.Timestamp occurred_at = 2;
  EventKind kind = 3; 

  oneof payload {
    CreateDebate    create_debate    = 10;
    ClearDebates    clear_debates    = 11;
    DeleteDebate    delete_debate    = 12;
    EnterDebate     enter_debate     = 13;
    GoHome          go_home          = 14;
  }
}
